### ISSUE TITLE
can't see resource when model's policy before() returns false and viewAny() returns true

### ISSUE THREAD

**Issue by jdssem** [laravel/nova-issues#1771]
- Laravel Version: 5.8.28
- Nova Version: 2.0.9
- PHP Version: 7.2
- Operating System and Version: MacOS 10.14.6

I have both `User` model and `UserPolicy`. I also have `before()` method in that policy for administrators like below.

```
  public function before($user, $ability)
  {
      return $user->isAdmin();
  }
```

The problem is that when I set `before()` method like this, and the user is not the administrator, then I can't see the resource at all even if `viewAny()` returns `true`.

In short,
when you set `before()` returns `false`, and `viewAny()` returns `true` in a policy class, you can't see the associated resource.

I want my subgroup of users who is not the administrator to see or not to see resources by it's policy.

**Comment by niekdemelker** [comment]
This is expected behavour. When the `before()` returns false, the gate returns false and so the `viewAny()` will never be checked. If the before returns true, the gate returns true.

If the `before()` returns null it will continue to check the `viewAny()`

I guess if you change your function to this is will work.
```
public function before($user, $ability)
{
    if ($user->isAdmin()) {
        return true;
    }
}
```

**Comment by jbrooksuk** [comment]
I believe this is the correct behaviour, though I can’t tell this from the tests. @davidhemphill can you confirm? Once we know, I’ll get update the tests.

**Comment by niekdemelker** [comment]
@jbrooksuk I assume the before function is a `Gate::before()` as there is no documentation of a before function in the policy class, This counts for the framework docs and the Nova docs.

As seen in Laravel in `Illuminate\Auth\Access\Gate::raw`, if the before returns `null`, the regular callbacks are triggered. So if the before() returns a boolean, the viewAny() will never be called.

```
    /**
     * Get the raw result from the authorization callback.
     *
     * @param  string  $ability
     * @param  array|mixed  $arguments
     * @return mixed
     */
    public function raw($ability, $arguments = [])
    {
        $arguments = Arr::wrap($arguments);

        $user = $this->resolveUser();

        // First we will call the "before" callbacks for the Gate. If any of these give
        // back a non-null response, we will immediately return that result in order
        // to let the developers override all checks for some authorization cases.
        $result = $this->callBeforeCallbacks(
            $user, $ability, $arguments
        );

        if (is_null($result)) {
            $result = $this->callAuthCallback($user, $ability, $arguments);
        }

        // After calling the authorization callback, we will call the "after" callbacks
        // that are registered with the Gate, which allows a developer to do logging
        // if that is required for this application. Then we'll return the result.
        return $this->callAfterCallbacks(
            $user, $ability, $arguments, $result
        );
    }
```

**Comment by jdssem** [comment]
@niekdemelker  Thanks for your help. I didn't know that If null is returned, the authorization will fall through to the policy method. I've just read the documentation too.

**Comment by Christophvh** [comment]
> @niekdemelker Thanks for your help. I didn't know that If null is returned, the authorization will fall through to the policy method. I've just read the documentation too.

Was struggling with this. Can this be documented somewhere that returning null will work as expected and false not? Because doing a gate Before in a permission system is fairly common we always use it for our super-admin roles.
